nemo/collections/asr/parts/perturb.py

```diff
@@ -74,9 +75,10 @@ def read_one_audiosegment(manifest, target_sr, rng, tarred_audio=False, audio_da
     else:
```

Comment:
Above here, in the if case, update the value of duration from the manifest entry

Reply:
yes, this I missed, added now for tarred dataset case

Reply:
Looks good. Let @jbalam-nv review and then merge when ready.

Reply:
@nithinraok so this was causing your OOM errors? In my tests, I did not use duration as I split the noise files into 20s chunks. Looks good. Could you add a test for this :).

Reply:
yes with these changes, runs looking good for now :)